scripts/vault_login.py

```diff
@@ -0,0 +1,100 @@
+import json
+import os
+import re
+import subprocess
+import sys
+import time
+
+from robottelo.errors import InvalidVaultURLForOIDC
+
+HELP_TEXT = (
+    "Vault CLI in not installed in your system, "
+    "refer link https://learn.hashicorp.com/tutorials/vault/getting-started-install to "
+    "install vault CLI as per your system spec!"
+)
+
+
+def _load_env():
+    print("\nFetching vault data from .env file and Exporting vault URL .....\n")
+    ROOT_PATH = os.path.realpath(
+        os.path.join(os.path.dirname(os.path.abspath(__file__)), os.path.pardir)
+    )
+    with open(ROOT_PATH + '/.env') as envfile:
+        envdata = envfile.read()
+    return ROOT_PATH, envdata
+
+
+def _export_vault_url(envdata):
+    vaulturl = re.findall('VAULT_URL_FOR_DYNACONF=(.*)', envdata)[0]
+    if 'localhost:8200' in vaulturl:
+        raise InvalidVaultURLForOIDC(
+            f"\033[3;31m{vaulturl} doesnt supports OIDC login,"
+            "please change url to corp vault in env file!"
+        )
+    os.environ['VAULT_ADDR'] = vaulturl
+
+
+def _vault_command(command: str):
+    vcommand = subprocess.run(command, capture_output=True, shell=True)
+    if vcommand.returncode != 0:
+        verror = str(vcommand.stderr)
+        if 'vault: command not found' in verror:
+            print(f"\033[1;31mError! {HELP_TEXT}")
+        elif 'Error revoking token' in verror:
+            print("\033[1;32mToken is alredy revoked!")
+        elif 'Error looking up token' in verror:
+            print("\033[1;33mWarning! Vault not logged in, please run 'make vault-login'!")
+        else:
+            print(f"\033[1;31mError! {verror}")
+        sys.exit(0)
+    return vcommand
+
+
+def _vault_login(root_path, envdata):
+    print(
+        "\033[1;30mWarning! The browser is about to open for vault OIDC login, "
+        "close the tab once the sign in is done!"
+    )
+    time.sleep(5)
+    if _vault_command(command="vault login -method=oidc").returncode == 0:
+        os.system("vault token renew -i 10h")
+        print("\033[1;32mSuccess! Vault OIDC Logged-In and extended for 10 hours!")
+    # Fetching token
+    token = _vault_command("vault token lookup --format json").stdout
+    token = json.loads(str(token.decode('UTF-8')))['data']['id']
+    # Setting new token in env file
+    envdata = re.sub('.*VAULT_TOKEN_FOR_DYNACONF=.*', f"VAULT_TOKEN_FOR_DYNACONF={token}", envdata)
+    with open(root_path + '/.env', 'w') as envfile:
+        envfile.write(envdata)
+    print("\033[1;32mSuccess! New OIDC token is added to .env file to access secrets from vault!")
+
+
+def _vault_logout(root_path, envdata):
+    # Teardown - Setting dymmy token in env file
+    envdata = re.sub('.*VAULT_TOKEN_FOR_DYNACONF=.*', "# VAULT_TOKEN_FOR_DYNACONF=myroot", envdata)
+    with open(root_path + '/.env', 'w') as envfile:
+        envfile.write(envdata)
+    vlogout = _vault_command('vault token revoke -self')
+    if vlogout.returncode == 0:
```

Comment:
These checks against returncode are redundant, `_vault_command` will `sys.exit()` if its non-zero.

Reply:
Removed the check!